src/Promenade.Controllers/HomeController.cs

```diff
@@ -1,21 +1,362 @@
+            }
+            else
+            {
+                var viewModel = new Location();
+                viewModel.CloseLocations = locations.OrderBy( c => c.Name).ToList();
+                if (mobile==1)
+                {
+                    viewModel.ShowLocation = true;
+                }
+                return View("Locations", viewModel);
+            }
+        }
+
+        [HttpGet("[action]/{locationStub}")]
+        [HttpGet("[action]/{locationStub}/{featureStub}")]
+        public IActionResult Locations(string locationStub, string featureStub)
+        {
+            Location[] locations = {  };
+
+
+            Feature[] features = {  };
+
+
+            LocationFeature[] locationsFeat = {  };
+
+            var group = new Group { Id = 1, GroupType="distance" };
+
+
+            LocationGroup[] locagroups = { };
+
+
+            if (string.IsNullOrEmpty(locationStub))
+            {
+
+                return View("Locations",locations);
+            }
+            else if (string.IsNullOrEmpty(featureStub))
+            {
+                var locationViewModel = new LocationViewModel();
+
+                foreach (var location in locations)
+                {
+                    if (location.Stub == locationStub)
+                    {
+                        locationViewModel.Location = location;
+                        var featlist = new List<LocationsFeaturesViewModel>();
+                        var neighbors = new List<Location>();
+                        foreach (var item in locationsFeat)
+                        {
+                            if (item.LocationId == location.Id)
+                            {
+                                foreach (var feat in features)
+                                {
+                                    if (item.FeatureId == feat.Id)
+                                    {
+                                        var locafeat = new LocationsFeaturesViewModel
+                                        {
+                                            Text = item.Text,
+                                            Stub = feat.Stub,
+                                            Name = feat.Name,
+                                            ImagePath = feat.ImagePath,
+                                            FontAwesome = feat.FontAwesome,
+                                            BodyText = feat.BodyText,
+                                            RedirectUrl = feat.Name == "Facebook"
+                                            ? location.Facebook
+                                            : item.RedirectUrl
+                                        };
+                                        featlist.Add(locafeat);
+                                    }
+                                }
+                            }
+                        }
+                        var groupid = locagroups.First(c => c.LocationId == location.Id && c.GroupId == 1).GroupId;
```

Comment:
1. Rename this to `groupId`.
2. I believe `.First()` will throw an exception if there's no match which is problematic. If you change it to `.FirstOrDefault()` it will return null if there's no match but then your `.GroupId` will throw an exception. Handle null cases here more gracefully.